core/node/api_server/src/tx_sender/tests/send_tx.rs

```diff
@@ -298,3 +308,75 @@ async fn sending_transaction_out_of_gas() {
     let (_, vm_result) = tx_sender.submit_tx(tx, block_args).await.unwrap();
     assert_matches!(vm_result.result, ExecutionResult::Revert { .. });
 }
+
+#[tokio::test]
+async fn submitting_tx_with_validation_traces() {
```

Comment:
Bikeshedding: IMO, this test is mostly useless; it essentially tests that the mock tx executor works as expected. I'd advise to test the *real* executor (as most tests in this module do) with the changed custom account contract; so essentially, move the new integration tests here. Likewise, you can test gas estimation as well.